src/exceptions/exceptions.py

```diff
@@ -0,0 +1,42 @@
+"""
+exceptions.py
+
+Custom exception classes for the CS3 client.
+"""
+
+# standard file missing message
+ENOENT_MSG = "No such file or directory"
+
+# standard error thrown when attempting to overwrite a file/xattr in O_EXCL mode
+# or when a lock operation cannot be performed because of failed preconditions
+EXCL_ERROR = "File/xattr exists but EXCL mode requested, lock mismatch or lock expired"
+
+# standard error thrown when attempting an operation without the required access rights
+ACCESS_ERROR = "Operation not permitted"
+
+
+class AuthenticationException(Exception):
```

Comment:
It is used in both the auth class and the file class, in case one of the requests (e.g. StatRequest) returns code_unauthenticated, which technically shouldn't happen since we check the token before each call, but they could potentially return code_unauthenticated.